### ISSUE TITLE
Next/previous doesn't listen to with_scope

### ISSUE THREAD

**Issue by ulitiy** [locomotivecms/engine#1023]
Some dirty code to fix it:
https://gist.github.com/ulitiy/b2757646ae6102faa130

could pass context inside model, but doubted if it would be better than to copy with_scope logic here.
could create ContentTypeProxyCollection, but no context inside (could set it though)

Don't know what solution is better, but at least it works. It's necessery feature for example if you need next/prev inside subcategories of entries.

Maybe it would be better to keep next/prev and other liquid logic outside models. To make the engine more modular.

**Comment by did** [comment]
@ulitiy good point. The code has been moved to Steam (https://github.com/locomotivecms/steam/blob/master/lib/locomotive/steam/repositories/content_entry_repository.rb#L87). 
That should not be too hard to pass a conditions hash to the next/previous methods and wire everything with the matching liquid drop class.
